File: twosumII.py
from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        hash: dict = {}
        n: int = len(numbers)
        for i in range(n):
            other: int = target - numbers[i]
            if other in hash:
                return [hash[other] + 1, i + 1]
            hash[numbers[i]] = i


sol = Solution()
print(sol.twoSum([2, 7, 11, 15], 9))

File: c++/twosumII.cpp
class Solution {
public:
  vector<int> twoSum(vector<int> &numbers, int target) {
    int left = 0;
    int right = numbers.size() - 1;
    int remain;
    while (left < right) {
      remain = target - numbers[right];
      if (remain == numbers[left]) {
        return {left + 1, right + 1};
      } else if (remain < numbers[left]) {
        right--;
      } else {
        left++;
      }
    }
    return {left + 1, right + 1};
  }
};
